sympy/ntheory/residue_ntheory.py

```diff
@@ -1196,10 +1195,12 @@ def _discrete_log_pollard_rho(n, a, b, order=None, retries=10, rseed=None):
 
             if xa == xb:
                 r = (ba - bb) % order
-                if r != 0:
-                    return mod_inverse(r, order) * (ab - aa) % order
-                break
```

Comment:
It is possible that this `break` should be left after the new code (at the same indentation). It will exit the inner loop and repeat the outer retry loop  with new `aa` and `ba`. That may save some time when the numbers are big and there is no solution. You could try to find such an example and compare the times. (Though I do not expect the difference be very important in general.)

Reply:
Thanks, I think the difference can be large I've tried with our initial example `_discrete_log_pollard_rho(11, 7, 21, rseed=0)` and the number of iterations without break was almost 3.5 times more i.e 35 than the number of iterations with break i.e 10. So, if for such small values of n, a, b we have that much difference hence it can be huge for bigger values of n, a, and b.